R/commonMachineLearningClustering.R

```diff
@@ -619,3 +619,13 @@
     }
   }
 }
+
+.mlClusteringCalculateDistances <- function(x) {
+  p <- try({
+    distx <- dist(x) # This scales terribly in terms of memory (O(n^2))
+  })
+  if (isTryError(p)) {
+    jaspBase:::.quitAnalysis(gettext("An error occurred in the analysis: You have run out of usable memory while calculating the distance matrix."))
```

Comment:
Good idea to centralize the `dist` computation!

I would be a bit more precise here than "memory". Perhaps something like:
```r
gettextf("Insufficient RAM available to compute the distance matrix. The analysis tried to allocate %s Gb", .extractMemSizeFromError(p))
```

because the user from the issue also mixed up RAM and disk space. It would also make sense to ensure that the error is due to insufficient RAM and only then show this message. When the error has another cause, we can just show the original error message. Showing the original error message will probably only happen when a programmer messes up, but if we don't then the error message about insufficient memory will be very misleading.

Reply:
Good idea, I've implemented this suggestion!